src/app/features/home/details/details.page.ts

```diff
@@ -454,6 +524,65 @@ export class DetailsPage {
       .subscribe();
   }
 
+  private async handleDownloadC2paAction(diaBackendAsset: DiaBackendAsset) {
+    const filePath = await this.diaBackendAssetRepository
+      .downloadC2pa$(diaBackendAsset.id)
+      .pipe(
+        catchError((err: unknown) => {
+          if (err instanceof HttpErrorResponse) {
+            const errorMessage = err.error?.error?.message;
+            if (errorMessage) {
+              return this.errorService.toastError$(errorMessage);
+            }
+          }
+          return this.errorService.toastError$(err);
+        }),
+        switchMap(c2paResult =>
+          defer(() => {
+            let filePath = c2paResult.cid;
+            const ext = extension(diaBackendAsset.asset_file_mime_type);
+            if (ext) {
+              filePath += `.${ext}`;
+            }
+            return Filesystem.downloadFile({
+              url: c2paResult.url,
+              path: filePath,
+              directory: Directory.Cache,
+            });
+          })
+        ),
+        catchError((err: unknown) => this.errorService.toastError$(err)),
+        switchMap(downloadResult => {
+          if (!downloadResult.path) {
+            return this.errorService.toastError$(
+              this.translocoService.translate('error.unknownError')
+            );
+          }
+          return of(downloadResult.path);
+        }),
+        untilDestroyed(this),
+        takeUntil(this.shareMemuDismissed$)
+      )
+      .toPromise();
+
+    if (!filePath || !(await ShareService.canShare())) {
+      // Failed or web no navigator share
+      return;
+    }
+
+    let fileUrl = filePath;
+    if (!fileUrl.startsWith('file://')) {
+      fileUrl = `file://${fileUrl}`;
+    }
+    await ShareService.shareFile(fileUrl)
+      .catch(async reason => {
+        if (reason?.message !== 'Share canceled') {
+          await this.errorService.toastError$(reason).toPromise();
+        }
+      })
+      .finally(() => Filesystem.deleteFile({ path: filePath }));
+  }
```

Comment:
Is it possible that on some sharing target platforms, shared files (especially image) are compressed and lost the C2PA injection data? As far as I know it happens at least on Slack and some major social media platforms.

Upload an C2PA image to slack -> download uploaded file -> get a file with different checksum and no C2PA claim

Would such situation happen when using the `shareFile` function? If yes we could probably consider sharing the IPFS url or add some warning about possible target platform compression.

Reply:
There is a [strip](https://techcrunch.com/2020/05/11/slack-strips-location-data/) metadata mechanism on Slack. However, since the URL has just been added but not pinned to IPFS, I think it's better to download it immediately to prevent it from being garbage collected. Additionally, besides sharing on social media, the function can also be shared to app files or emails, so social media might be considered a limitation. I'll ask about it in the task, and we could update it in the future sprint if the UI or text should be modified.